cmd/kube-controller-manager/app/options/options.go

```diff
@@ -46,6 +46,7 @@ func NewCMServer() *CMServer {
 			Port:                              ports.ControllerManagerPort,
 			Address:                           "0.0.0.0",
 			ConcurrentEndpointSyncs:           5,
+			ConcurrentServiceSyncs:            5,
```

Comment:
Yeah it's reasonable, but introduces a concurrency issue when we don't really need to optimize this component for performance. 

I'm actually not 100% confident there were races before, because there were situations where we'd get N updates, each would spin off a goroutine, they'd all wait on a lock, and wake up and take the lock in random order. This means out of order updates, which breaks correctness. 

I believe such issues should be fixed with the workqueue. But why multithread when we can avoid it?